sycl/test-e2e/Matrix/SG32/joint_matrix_bfloat16_accumulator.cpp

```diff
@@ -0,0 +1,32 @@
+//==-------- SG32/joint_matrix_bfloat16_accumulator.cpp  - DPC++ joint_matrix----------- ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// SG size = 32 is not currently supported for SYCL Joint Matrix by IGC on DG2
+// UNSUPPORTED: gpu-intel-dg2
+
+// UNSUPPORTED: target-nvidia, target-amd
+// UNSUPPORTED-INTENDED: aspect-ext_intel_matrix isn't currently supported for
+// other triples
+
+// REQUIRES: aspect-ext_intel_matrix
+// REQUIRES-INTEL-DRIVER: lin: 27501, win: 101.4943
+
+// XFAIL: gpu
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// RUN: %if gpu %{ env IGC_JointMatrixLoadStoreOpt=2 %{run} %t.out %}
+// RUN: %if gpu %{ env IGC_JointMatrixLoadStoreOpt=1 %{run} %t.out %}
+// RUN: %if gpu %{ env IGC_JointMatrixLoadStoreOpt=0 %{run} %t.out %}
+
+
+#include "common.hpp"
+
+#define SG_SZ 32
+#define ACC_BFLOAT16 true
+
+#include "joint_matrix_bfloat16_impl.hpp"
```

Comment:
with setup like that, the test is just repeating `sycl/test-e2e/Matrix/SG32/joint_matrix_bfloat16.cpp`, based on how macro logic is implemented in the .hpp file. So, I guess it should pass actually.
the only difference is it runs with all 3 `IGC_JointMatrixLoadStoreOpt` settings.
But it doesn't verify bfloat16 accumulator actually, since all that code is under !SG32 macro in .hpp file.

If the point of the test is to run shapes with bfloat16 accumulator under sub-group size 32 and have it xfail, until this is implemented, then running of those specific tests should be moved into this file, from .cpp file, similar to sycl/test-e2e/Matrix/joint_matrix_out_bounds_big_shapes.cpp